Sources/OpenAPIRuntime/Interface/CurrencyTypes.swift

```diff
@@ -11,281 +11,100 @@
+        return path[path.startIndex..<pathEndIndex]
     }
 }
 
-/// An HTTP request, sent by the client to the server.
-public struct Request: Hashable, Sendable {
+extension HTTPResponse {
 
-    /// The path of the URL for the HTTP request.
-    public var path: String
-
-    /// The query string of the URL for the HTTP request.
-    ///
-    /// A query string provides support for assigning values to parameters
-    /// within a URL.
-    ///
-    /// _URL encoding_, officially known as _percent-encoding_, is a method
-    /// to encode arbitrary data in a URI using only ASCII characters.
-    ///
-    /// An example of a URL with a query string is:
-    ///
-    /// ```
-    /// https://example.com?name=Maria%20Ruiz&email=mruiz2%40icloud.com
-    /// ```
-    ///
-    /// For this request, the query string is:
-    ///
-    /// ```
-    /// name=Maria%20Ruiz&email=mruiz2%40icloud.com
-    /// ```
-    ///
-    /// - NOTE: The `?` is a seperator in the URL and is **not** part of
-    /// the query string.
-    /// - NOTE: Only query parameter names and values are percent-encoded,
-    /// the `&` and `=` remain.
-    public var query: String?
-
-    /// The method of the HTTP request.
-    public var method: HTTPMethod
-
-    /// The header fields of the HTTP request.
-    public var headerFields: [HeaderField]
-
-    /// The body data of the HTTP request.
-    public var body: Data?
-
-    /// Creates a new HTTP request.
+    /// Creates a new response.
     /// - Parameters:
-    ///   - path: The path of the URL for the request. This must not include
-    ///   the base URL of the server.
-    ///   - query: The query string of the URL for the request. This should not
-    ///   include the separator question mark (`?`) and the names and values
-    ///   should be percent-encoded. See ``query`` for more information.
-    ///   - method: The method of the HTTP request.
-    ///   - headerFields: The header fields of the HTTP request.
-    ///   - body: The body data of the HTTP request.
-    ///
-    /// An example of a request:
-    /// ```
-    /// let request = Request(
-    ///   path: "/users",
-    ///   query: "name=Maria%20Ruiz&email=mruiz2%40icloud.com",
-    ///   method: .GET,
-    ///   headerFields: [
-    ///       .init(name: "Accept", value: "application/json"
-    ///   ],
-    ///   body: nil
-    /// )
-    /// ```
-    public init(
-        path: String,
-        query: String? = nil,
-        method: HTTPMethod,
-        headerFields: [HeaderField] = [],
-        body: Data? = nil
-    ) {
-        self.path = path
-        self.query = query
-        self.method = method
-        self.headerFields = headerFields
-        self.body = body
+    ///   - statusCode: The status code of the response.AsString
+    ///   - headerFields: The HTTP header fields.
+    public init(soar_statusCode statusCode: Int, headerFields: HTTPFields = .init()) {
```

Comment:
Same as https://github.com/apple/swift-openapi-runtime/pull/47#discussion_r1336746165, I'm trying to avoid a potential future naming conflict as this is extending a type we don't own.